File: CarServiceGame/CarServiceGame.Domain/Entities/RepairProcess.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CarServiceGame.Domain.Entities
{
    public class RepairProcess
    {
        public RepairOrder Order { get; set; }
        public Worker AssignedWorker { get; set; }
        public int StallNumber { get; set; }
        public DateTime CreatedDate { get; set; }

        public RepairProcess() { }

        public RepairProcess(RepairOrder order, Worker worker)
        {
            if(order == null || worker == null)
                throw new ArgumentException("order or worker is null");

            Order = order;
            AssignedWorker = worker;
        }

        public int GetRequiredTime()
        {
            return  Order.RequiredWork/AssignedWorker.Efficiency - (int) DateTime.Now.Subtract(CreatedDate).TotalSeconds;
        }
    }
}

File: CarServiceGameWeb/src/app/model/repair-process.ts
import { Order } from "./order";
import { Worker } from "./worker";

export class RepairProcess {
    private _order: Order;
    private _assignedWorker: Worker;
    private _stallNumber: number;
    private _createdDate: Date;
    private _isCancelled: boolean;


	constructor(order: Order, assignedWorker: Worker, stallNumber: number, createdDate: Date, isCancelled: boolean) {
		this._order = order;
		this._assignedWorker = assignedWorker;
		this._stallNumber = stallNumber;
		this._createdDate = createdDate;
		this._isCancelled = isCancelled;
	}
    

	public get Order(): Order {
		return this._order;
	}

	public set Order(value: Order) {
		this._order = value;
	}

	public get AssignedWorker(): Worker {
		return this._assignedWorker;
	}

	public set AssignedWorker(value: Worker) {
		this._assignedWorker = value;
	}

	public get StallNumber(): number {
		return this._stallNumber;
	}

	public set StallNumber(value: number) {
		this._stallNumber = value;
	}

	public get CreatedDate(): Date {
		return this._createdDate;
	}

	public set CreatedDate(value: Date) {
		this._createdDate = value;
	}

	public get IsCancelled(): boolean {
		return this._isCancelled;
	}

	public set IsCancelled(value: boolean) {
		this._isCancelled = value;
	}
    
}